packages/@react-aria/interactions/src/useFocusVisible.ts

```diff
@@ -84,13 +84,35 @@ function handleClickEvent(e: MouseEvent) {
   }
 }
 
+function isEventTargetNegativeTabIndex(e: Event) {
+  if (!(e.target instanceof Element)) {
+    return false;
+  }
+
+  const tabIndex = e.target.getAttribute('tabIndex');
+
+  if (tabIndex === null) {
+    return false;
+  }
+
+  const numberedTabIndex = parseInt(tabIndex, 10);
+  return isNaN(numberedTabIndex) ? false : numberedTabIndex < 0;
+}
+
 function handleFocusEvent(e: FocusEvent) {
   // Firefox fires two extra focus events when the user first clicks into an iframe:
   // first on the window, then on the document. We ignore these events so they don't
   // cause keyboard focus rings to appear.
   if (e.target === window || e.target === document) {
     return;
   }
+  // Elements with a negative tabIndex will fire focus events. In Chrome and Firefox, this can happen briefly
```

Comment:
Is it really specific to _negative_ tabIndex? Wouldn't any element with a tabIndex be focusable? What exactly is the sequence of events leading to the bad outcome?